src/etc/inc/pfsense-utils.inc

```diff
@@ -3562,4 +3562,13 @@ function send_user_download($type = 'data', $content, $userfilename = "", $conte
 	exit;
 }
 
+// Test whether the hostname in a URL can be resolved with a very short timeout
+function is_url_hostname_resolvable($url) {
+	$urlhostname = explode("/", $url)[2];
```

Comment:
Assuming that will return an array with 3+ parts is dangerous. You should be using [parse_url()](https://www.php.net/manual/en/function.parse-url.php) and checking for the appropriate field in the results.